includes/plugins/class-woocommerce-gateway-stripe.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * WooCommerce Gateway Stripe integration class.
+ * https://wordpress.org/plugins/woocommerce-gateway-stripe
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Main class.
+ */
+class WooCommerce_Gateway_Stripe {
+	/**
+	 * Initialize hooks and filters.
+	 */
+	public static function init() {
+		add_filter( 'wc_stripe_intent_metadata', [ __CLASS__, 'add_transaction_metadata' ], 10, 2 );
+	}
+
+	/**
+	 * Add metadata to a Stripe transaction.
+	 *
+	 * @param array    $metadata Array of keyed metadata values.
+	 * @param WC_Order $order Order being processed.
+	 *
+	 * @return array Array of keyed metadata values.
+	 */
+	public static function add_transaction_metadata( $metadata, $order ) {
+		// Skip orders with multiple products.
+		if ( $order->get_item_count() > 1 ) {
+			return $metadata;
+		}
+
+		$order_item = array_values( $order->get_items() )[0];
+		if ( ! $order_item ) {
+			return $metadata;
+		}
+		$product_id = $order_item->get_product_id();
+
+		// Product name.
+		$metadata[ __( 'Product', 'newspack-plugin' ) ] = $order_item->get_name();
```

Comment:
I know we are migrating this existing feature from another repo... but I'm not sure it's a good idea to internationalize the metadata key... 

* a wrong translation can bring unsupported characters or strings that are too long
* If they change the language they'll have data in both formats
* harder to provide support and identify which field is which
* Everything in Stripe is in english anyways, isnt it?

Reply:
Huh, good call! Updated